pkg/storage/store.go

```diff
@@ -2690,35 +2695,35 @@ func (s *Store) maybeWaitInPushTxnQueue(
 
 // reserveSnapshot throttles incoming snapshots. The returned closure is used
 // to cleanup the reservation and release its resources. A nil cleanup function
-// and a nil error indicates the reservation was declined.
+// and a non-empty rejectionMessage indicates the reservation was declined.
 func (s *Store) reserveSnapshot(
 	ctx context.Context, header *SnapshotRequest_Header,
-) (func(), error) {
+) (cleanup func(), rejectionMsg string, err error) {
```

Comment:
You named these purely to make the signature easier to understand, right?
I wonder how our linters like `(_cleanup func(), _rejectionMsg string, _err error)` for this kind of thing (that's the way it works in Rust for avoiding making vars unavailable but at the same time explaining them). If that doesn't work and you feel so inclined, maybe you don't have to name the `err` return.

Reply:
Putting underscores in front of them is certainly an interesting approach. It may be the only place we ever end up doing this in our code, but it's a standard I'd be happy to see us move to.